flink-staging/flink-streaming/flink-streaming-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSource.java

```diff
@@ -89,12 +99,16 @@ public void run(SourceContext<OUT> ctx) throws Exception {
 		while (running) {
 			delivery = consumer.nextDelivery();
 
-			OUT result = schema.deserialize(delivery.getBody());
-			if (schema.isEndOfStream(result)) {
-				break;
+			synchronized (ctx.getCheckpointLock()) {
+				OUT result = schema.deserialize(delivery.getBody());
+				addId(ctx, delivery.getEnvelope().getDeliveryTag());

```

Comment:
This looks like it will grow the id array infinitely when the checkpointing is disabled.